src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -365,6 +365,17 @@ pub(crate) fn get_tool_rustc_compiler(
     builder.compiler(target_compiler.stage.saturating_sub(1), builder.config.host_target)
 }
 
+/// Returns a compiler that is able to compile a `ToolTarget` tool for the given `target`.
+pub(crate) fn get_tool_target_compiler(builder: &Builder<'_>, target: TargetSelection) -> Compiler {
+    todo!("FIX");
+    if builder.host_target == target {
+        builder.compiler(0, builder.host_target)
```

Comment:
Question [STAGING 1/2]: wait, does this actually agree with the comment

> While we could compile them using the stage0 compiler when not cross-compiling, we instead
> use the in-tree compiler (and std) to build them, so that we can ship e.g. std security
> fixes and avoid depending fully on stage0 for the artifacts that we ship.

are we not using the stage 0 compiler (not in-tree!) when not cross-compiling?